database/skylink.go

```diff
@@ -72,34 +82,83 @@ func (db *DB) SkylinkFetch(ctx context.Context, sl string) (Skylink, error) {
 		return Skylink{}, sr.Err()
 	}
 	s := Skylink{}
-	err := sr.Decode(&s)
+	err = sr.Decode(&s)
 	if err != nil {
 		return Skylink{}, err
 	}
 	return s, nil
 }
 
+// SkylinkMarkPinned marks a skylink as pinned (or no longer unpinned), meaning
+// that Pinner should make sure it's pinned by the minimum number of servers.
+func (db *DB) SkylinkMarkPinned(ctx context.Context, skylink string) error {
+	sl, err := CanonicalSkylink(skylink)
+	if err != nil {
+		return ErrInvalidSkylink
+	}
+	filter := bson.M{"skylink": sl}
+	update := bson.M{"$set": bson.M{"unpin": false}}
+	opts := options.UpdateOptions{}
+	opts.SetUpsert(true)
+	_, err = db.staticDB.Collection(collSkylinks).UpdateOne(ctx, filter, update)
+	return err
+}
+
+// SkylinkMarkUnpinned marks a skylink as unpinned, meaning that all servers
+// should stop pinning it.
+func (db *DB) SkylinkMarkUnpinned(ctx context.Context, skylink string) error {
+	sl, err := CanonicalSkylink(skylink)
+	if err != nil {
+		return ErrInvalidSkylink
+	}
+	filter := bson.M{"skylink": sl}
+	update := bson.M{"$set": bson.M{"unpin": true}}
+	opts := options.UpdateOptions{}
```

Comment:
just in case you didn't know about this, I don't mind the current format
`opts := options.Update().SetUpsert(true)`

Reply:
yes, I know about it and I'm suing it but it seems I'm inconsistent with it :)

btw, a nice side benefit of this approach is that if you forget to pass the `opts` to the actual `Update` your editor will mark it as unused while if you do `opts.SetUpsert(true)` it won't.